docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -82,30 +82,158 @@ a :ref:`connector-page-sink-provider`.
+simpler than implementing a ``RecordCursor``.
+
+A ``RecordCursor`` implementation needs to keep track of the current record.
+It return values for columns by a numerical position, in the data type matching
+the column definition in the table. When the engine is done reading the current
+record it will call ``advanceNextPosition`` on the cursor.
+
+Type mapping
+^^^^^^^^^^^^
+
+The built-in SQL data types use different Java types as carrier types.
+
+================================== =============
+SQL type                           Java type
+================================== =============
+``BOOLEAN``                        ``boolean``
+``TINYINT``                        ``long``
+``SMALLINT``                       ``long``
+``INTEGER``                        ``long``
+``BIGINT``                         ``long``
+``REAL``                           ``double``
+``DOUBLE``                         ``double``
+``DECIMAL``                        ``long`` for precision up to 19, inclusive;
+                                   ``Int128`` for precision greater than 19
+``VARCHAR``                        ``Slice``
+``CHAR``                           ``Slice``
+``VARBINARY``                      ``Slice``
+``JSON``                           ``Slice``
+``DATE``                           ``long``
+``TIME(P)``                        ``long``
+``TIME WITH TIME ZONE``            ``long`` for precision up to 9;
+                                   ``LongTimeWithTimeZone`` for precision
+                                   greater than 9
+``TIMESTAMP(P)``                   ``long`` for precision up to 6;
+                                   ``LongTimestamp`` for precision greater
+                                   than 6
+``TIMESTAMP(P) WITH TIME ZONE``    ``long`` for precision up to 3;
+                                   ``LongTimestampWithTimeZone`` for precision
+                                   greater than 3
+``INTERVAL YEAR TO MONTH``         ``long``
+``INTERVAL DAY TO SECOND``         ``long``
+``ARRAY``                          ``Block``
+``MAP``                            ``Block``
+``ROW``                            ``Block``
+``IPADDRESS``                      ``[]byte``
+``UUID``                           ``[]byte``
+``HyperLogLog``                    ``Object``
+``P4HyperLogLog``                  ``Object``
+``SetDigest``                      ``Object``
+``QDigest``                        ``Object``
+``TDigest``                        ``Object``
+================================== =============
+
+The ``RecordCursor.getType(int field)`` method returns the SQL type for a field
+and the value for it is returned by one of the following methods, matching
+the carrier type:
+
+* ``getBoolean(int field)``
+* ``getLong(int field)``
+* ``getDouble(int field)``
+* ``getSlice(int field)``
+* ``getObject(int field)``
+
+
+Date and time values of regular precision can be converted into ``long`` using
```

Comment:
This is only for `timestamp(p) with time zone` and `time(p) with time zone` types.